Add new BiBiB system

bobjacobsen opened the pull request:
This adds a new BiBiB system connection.  The code was contributed by Eckart Meyer

bobjacobsen commented:
WIP while we get the CI to pass and re-check any changes needed.

danielb987 commented on java/src/jmri/jmrix/bidib/BiDiBProgrammerManager.java:
```diff
@@ -0,0 +1,112 @@
+import jmri.managers.DefaultProgrammerManager;
+
+/**
+ * Extend DefaultProgrammerManager to provide ops mode programmers for NCE
+ * systems
```
No need to change, but this comment doesn't look right.

danielb987 commented on java/src/jmri/jmrix/bidib/BiDiBThrottle.java:
```diff
@@ -0,0 +1,539 @@
+
+
+/**
+ * An implementation of DccThrottle with code specific to an NCE connection.
```
No need to change, but this comment doesn't look right.

danielb987 commented on java/src/jmri/jmrix/bidib/BiDiBThrottleManager.java:
```diff
@@ -0,0 +1,146 @@
+import org.slf4j.LoggerFactory;
+
+/**
+ * NCE implementation of a ThrottleManager.
```
No need to change, but this comment doesn't look right.

danielb987 commented on java/src/jmri/jmrix/bidib/BiDiBTrafficController.java:
```diff
@@ -0,0 +1,1487 @@
+ * @author Eckart Meyer Copyright (C) 2019-2023
+ *
+ */
+@SuppressFBWarnings(value = "JLM_JSR166_UTILCONCURRENT_MONITORENTER")
```
What does this `@SuppressFBWarnings` actually suppress?

Quote from https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html
> JLM: Synchronization performed on util.concurrent instance (JLM_JSR166_UTILCONCURRENT_MONITORENTER)
> 
> This method performs synchronization on an object that is an instance of a class from the java.util.concurrent package (or its subclasses). Instances of these classes have their own concurrency control mechanisms that are orthogonal to the synchronization provided by the Java keyword synchronized. For example, synchronizing on an AtomicBoolean will not prevent other threads from modifying the AtomicBoolean.
> 
> Such code may be correct, but should be carefully reviewed and documented, and may confuse people who have to maintain the code at a later date.

bobjacobsen replied:
I've added a comment that discusses how these AtomicBooleans are being used for synchronization.

danielb987 commented on java/src/jmri/jmrix/bidib/BiDiBTurnoutManager.java:
```diff
@@ -0,0 +1,190 @@
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implement turnout manager for EasyDcc systems.
```
No need to change, but this comment doesn't look right.

danielb987 commented on java/src/jmri/jmrix/bidib/serialdriver/README:
```diff
@@ -0,0 +1,11 @@
+This is the README file for the jmrix.nce.serialdriver package.
+
+The jmrix Java package defines implementations for the JMRI interfaces. jmrix.nce.serialdriver
+defines classes for interfacing to an NCE command station via a RS232 serial line. 
```
No need to change, but this doesn't look right.

danielb987 reviewed (dismissed):
I found some minor issues but you can ignore them if you want to.

bobjacobsen commented:
Thanks for the careful review!  I've addresses them all, I think: I corrected the comments that point to other systems. I fixed the tabs in build.xml and the organization in default_lcf.xml.  See reply above about the AtomicBoolean comment. Thank you again.

bobjacobsen commented:
CI is failing on UTF8 characters in two German help pages.  I've started a discussion on jmri-developers to decide how to address this.

danielb987 commented:
@bobjacobsen 
If you want, feel free to tell Eckart Meyer that I'm impressed of this PR!
